src/essence/Tools/Isochrone/IsochroneTool.js

```diff
@@ -0,0 +1,539 @@
+            [140, 150, 198],
+            [136, 86, 167],
+            [129, 15, 124]
+        ]
+    ],
+
+    initialize: function() {
+        this.vars = L_.getToolVars('isochrone');
+
+        //add info for scaling low-res tiles
+        for(const dataType in this.vars.data) {
+            this.dataSources[dataType] = [];
+            for(const src of this.vars.data[dataType]) {
+                const {name, tileurl, minZoom, maxNativeZoom, resolution} = src;
+                let zoomOffset = 0;
+                switch(resolution) {
+                    case 256:
+                        zoomOffset = 0;
+                    break;
+                    case 128:
+                        zoomOffset = 1;
+                    break;
+                    case 64:
+                        zoomOffset = 2;
+                    break;
+                    case 32:
+                        zoomOffset = 3;
+                    break;
+                    default:
+                        console.warn(`IsochroneTool: ${dataType} source "${src.name}" has undefined or nonstandard resolution: ${src.resolution}!`);
+                }
+                
+                const minResolution = Math.max(minZoom - zoomOffset, 0);
+                const maxResolution = maxNativeZoom - zoomOffset;
+
+                //Data source interface! In case this tool gets ts-ified
+                this.dataSources[dataType].push({
+                    name, //string
+                    tileurl, //string
+                    minZoom, //number
+                    maxNativeZoom, //number
+                    resolution, //number
+                    minResolution, //number
+                    maxResolution, //number
+                    zoomOffset //number
+                });
+            }
+        }
+    },
+
+    make: function() {
+        this.MMGISInterface = new interfaceWithMMGIS();
+        
+        $("#iscNew").on("click", () => this.addIsochrone());
+        this.containerEl = $("#isochroneOptionsContainer");
+        this.addIsochrone();
+    },
+
+    destroy: function() {
+        for(const {marker, layerName} of this.managers) {
+            Map_.rmNotNull(marker);
+            Map_.rmNotNull(L_.layersGroup[layerName]);
+        }
+
+        if(this.hoverPolyline !== null) {
+            this.hoverPolyline.remove(Map_.map);
+        }
+        
+        this.managers = [];
+        this.activeManager = null;
+        this.managerCounter = 0;
+        this.MMGISInterface.separateFromMMGIS();
+    },
+
+    getUrlString: function() { //TODO?
+        return '';
+    },
+
+    valueToColor: function(val, rampIndex, steps = 0) {
+        val = Math.min(val, 1);
+        if(steps) val = Math.floor(val * steps) / steps;
+
+        const ramp = this.colorRamps[rampIndex];
+        const color = val * (ramp.length - 1);
+        const i = Math.min(Math.floor(color), ramp.length - 2);
+        const off = color % 1;
+        const getChan = chan =>
```

Comment:
recommend using curly braces and `return` statements with multi-line arrow functions